lib/net.js

```diff
@@ -934,6 +935,7 @@ function internalConnect(
       localAddress = localAddress || '::';
       err = self._handle.bind6(localAddress, localPort);
     } else {
+      // eslint-disable-next-line no-restricted-syntax
       self.destroy(new TypeError('Invalid addressType: ' + addressType));
```

Comment:
This error doesn't have a code but I don't know what to do with it: Assign a code? Which one? How can we reach this error?

Reply:
I think you can provide a custom lookup function to `net.connect` that calls the callback with an invalid `addressType ` to trigger this. Something like

```js
net.connect({
  host: 'foo.bar'
  port: 80,
  lookup: (host, dnsopts, cb) => cb(null, '127.0.0.1', 100)  // 100 is not a valid addressType
})
```

Reply:
Also we probably need to invent a error code for this?

Reply:
Thanks. I guess there is something broken to fix first, because this is what I get with your example:
```
$ node test/parallel/test-net-connect-options-invalid-address-type.js
node[15295]: ../src/tcp_wrap.cc:282:static void node::TCPWrap::Connect(const v8::FunctionCallbackInfo<v8::Value>&): Assertion `args[2]->IsUint32()' failed.
 1: node::Abort() [node]
 2: 0x8cebdb [node]
 3: node::TCPWrap::Connect(v8::FunctionCallbackInfo<v8::Value> const&) [node]
 4: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&)) [node]
 5: 0xb1444c [node]
 6: v8::internal::Builtin_HandleApiCall(int, v8::internal::Object**, v8::internal::Isolate*) [node]
 7: 0x1f18982042fd
[1]    15295 abort (core dumped)  node test/parallel/test-net-connect-options-invalid-address-type.js
```

Reply:
This happens because `localAddress` and `localPort` are `undefined` so we end up in another path: 
https://github.com/targos/node/blob/244cdb1e033706756c0296d344cca5702dea261b/lib/net.js#L965-L971